eth/web3.c

```diff
@@ -235,3 +235,67 @@ int eth_getTransactionReceipt(web3_ctx_t *web3, const tx_hash_t *tx_hash)
     WEB3_TERMINATOR();
     return 0;
 }
+
+int eth_convert(const uint256_t *amount, enum ETH_UNIT from, enum ETH_UNIT to, char *buf, size_t buf_size)
+{
+    assert(amount != NULL);
+    size_t shift = from > to ? from - to : to - from;
+
+    if(tostring256(amount, 10, buf, buf_size) == false) {
+        return -1;
+    }
+
+    // special case '0'
+    if(buf[0] == '0') return 0;
+
+    // return if the conversion is the same unit
+    if (shift == 0) {
+        return 0;
+    // conversion to smaller unit, add '0'
+    } else if (from > to) {
+        size_t i;
+        size_t len = strlen(buf);
+        // check buffer size
+        if (len + 1 + shift > buf_size) {
+            return -1;
+        }
+        // add zeros
+        for (i = len; i < len + shift; ++i) {
+            buf[i] = '0';
+        }
+        buf[len + shift + 1] = '\0';
+    // conversion to bigger unit add '.'
+    } else {
+        size_t i;
+        size_t len = strlen(buf);
+        if (len  > shift) {
+            // check buffer size
+            if (len + 1 + 1 > buf_size) {
+                return -1;
+            }
+            // move numbers
+            for (i = len + 1 + 1; i > len - shift; --i) {
```

Comment:
One nitpick - could you replace magic numbers in string length calculation with defines? It will make it easier to read and understand.

i.e. here it would look like `for (i = len + DECIMAL_POINT + SOMETHING; i > len - shift; --i)`